sdk/ml/azure-ai-ml/azure/ai/ml/_schema/core/resource.py

```diff
@@ -40,10 +40,11 @@ def pass_source_path(self, data, original, **kwargs):
     def update_base_path_pre_dump(self, data, **kwargs):
         # inherit from parent if base_path is not set
         if data.base_path:
-            cur_base_path: Path = Path(data.base_path).resolve()
-            if not cur_base_path.samefile(self.context[BASE_PATH_CONTEXT_KEY]):
-                self._previous_base_path = Path(self.context[BASE_PATH_CONTEXT_KEY])
-                self.context[BASE_PATH_CONTEXT_KEY] = cur_base_path
+            cur_base_path: Path = Path(data.base_path)
```

Comment:
Did not get why we change this? Let's sync offline

Reply:
it's about the implementation of Path.samefile. It will try to resolve the 2 paths and then determine whether they point to the same file/directory. The good part is that, if `self` is a link to `other`, it will return `True`; the bad part is, if one of the base path is not an existent path, it will raise an Exception.

So here I made this change for 2 reasons:
1. actually we can always push base path here, so a soft condition to skip push bring no harm;
2. it will avoid resolving the base path in Schema. So even if the base path is a temp folder and is deleted before dumping, ValidationError will be raised and only raised when trying to dump a relative LocalPathField.